pdata/pprofile/attributes.go

```diff
@@ -0,0 +1,27 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofile // import "go.opentelemetry.io/collector/pdata/pprofile"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type attributable interface {
+	AttributeIndices() pcommon.Int32Slice
+}
+
+// BuildAttributes builds a [pcommon.Map] containing the attributes of a record.
+// The record can by any struct that implements an `AttributeIndices` method.
+func BuildAttributes(profile Profile, record attributable) (pcommon.Map, error) {
+	attrs := map[string]any{}
+
+	for i := range record.AttributeIndices().AsRaw() {
```

Comment:
Unnecessary slice allocation, avoid everywhere AsRaw and FromRaw and add a benchamark.

Reply:
Thank you. I've added the benchmark and switched to manual type setting. There's still 9 allocations though. Happy to look into any improvement proposal.